ambari/datadog_checks/ambari/ambari.py

```diff
@@ -0,0 +1,198 @@
+            raise CheckException(
+                "Couldn't connect to URL: {}. Please verify the address is reachable".format(clusters_endpoint))
+
+        self._submit_service_checks("can_connect", self.OK, ["url:{}".format(base_url)])
+        return [cluster.get('Clusters').get('cluster_name') for cluster in resp.get('items')]
+
+    def get_host_metrics(self, base_url, authentication, clusters, base_tags):
+        for cluster in clusters:
+            cluster_tag = CLUSTER_TAG_TEMPLATE.format(cluster)
+            hosts_list = self._get_hosts_info(base_url, authentication, cluster)
+
+            for host in hosts_list:
+                if host.get(METRICS_FIELD) is None:
+                    self.log.warning("No metrics received for host {}".format(host.get('Hosts').get('host_name')))
+                    continue
+
+                metrics = self.flatten_host_metrics(host.get(METRICS_FIELD))
+                for metric_name, value in iteritems(metrics):
+                    host_tag = HOST_TAG + host.get('Hosts').get('host_name')
+                    metric_tags = base_tags + [cluster_tag, host_tag]
+                    if isinstance(value, float):
+                        self._submit_gauge(metric_name, value, metric_tags)
+                    else:
+                        self.log.warning("Encountered non float metric {}:{}".format(metric_name, value))
+
+    def get_service_metrics(self, base_url, authentication, clusters, whitelisted_services,
+                            whitelisted_metrics, base_tags):
+        for cluster in clusters:
+            tags = base_tags + [CLUSTER_TAG_TEMPLATE.format(cluster)]
+            for service, components in iteritems(whitelisted_services):
+                service_tags = tags + [SERVICE_TAG + service.lower()]
+                self.get_component_metrics(base_url, authentication, cluster, service,
+                                           service_tags, [c.upper() for c in components], whitelisted_metrics)
+                self.get_service_checks(base_url, authentication, cluster, service, service_tags)
+
+    def get_service_checks(self, base_url, authentication, cluster, service, service_tags):
+        service_info = self._get_service_checks_info(base_url, authentication, cluster, service, service_tags)
+        for info in service_info:
+            self._submit_service_checks("state", info['state'], info['tags'])
+
+    def get_component_metrics(self, base_url, authentication, cluster, service,
+                              base_tags, component_whitelist, metric_whitelist):
+        component_metrics_endpoint = common.create_endpoint(base_url, cluster, service, COMPONENT_METRICS_QUERY)
+        components_response = self._make_request(component_metrics_endpoint, authentication)
+
+        if components_response is None:
+            self.log.warning("No components found for service {}.".format(service))
+            return
+
+        for component in components_response.get('items'):
```

Comment:
If there is no `items`, it will fail. Either it's always present, or you need to pass a default to `get`.

Reply:
I've added a check above